samples/sample_definition_and_criteria.rst

```diff
@@ -0,0 +1,104 @@
+
+    * Must not use :kconfig:option:`CONFIG_ZTEST` or :kconfig:option:`CONFIG_ZTEST_NEW_API`.
+    * Must not use zasserts in samples.
+
+  * If a sample can provide output that can be verified, then output should be evaluated against
+    expected value to have some level of testing for the sample itself.
+  * Samples are optional.
+
+2. Samples should target multiple platforms and architectures.
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+  * Minimize the use of ``platform_allow`` and architecture filters as it limits the scope
+    of testing to the mentioned platforms and architectures. Reference: `Testcase Docs`_
+  * Make use of ``integration_platforms`` to limit the scope when there are timing or
+    resource constraints.
+  * Make the sample as generic as possible. Avoid making a sample platform specific unless it is
+    for particular hardware.
+
+  .. _Testcase Docs: https://docs.zephyrproject.org/latest/develop/test/twister.html?highlight=test%20runner#test-cases.
+
+3. Twister should be able to build every sample.
+++++++++++++++++++++++++++++++++++++++++++++++++
+  * Every sample must have a YAML file.
+
+    **For example:**
+
+    .. code-block:: yaml
+
+      tests:
+        sample.kernel.cond_var:
+          integration_platforms:
+            - native_posix
+          tags: kernel condition_variables
+          harness: console
+          harness_config:
+            type: one_line
+            regex:
+              - ".*Waited and joined with 3 threads"
+
+  **Guidelines for Samples:**
+    * Minimize the use of ``platform_allow`` and architecture filters.
+    * Do not mark the test as build only unless necessary.
+    * Any test case-specific configuration options are added using ``extra_args`` or
+      ``extra_configs`` options in the YAML file. The prj.conf file should have the in-common
+      configuration options.
+    * The sample itself can be evaluated using multiple configurations in the sample's YAML file.
+      For example, the sample is used to run with different logging modes using multiple scenarios
+      in samples/subsys/logging/syst.
+    * Sample output can be validated leveraging the ``harness_config`` regex option,
+      wherever applicable.
+
+4. A sample should provide sufficient coverage of a subsystem, feature, or module.
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+  **DO's:**
+    * Cover the most common and important use cases of the functionality.
+    * Keep the code simple and easy to read. Example: samples/philosophers.
+
+  **DONT's:**
+    * Samples must not test the negative or edge case behaviors.
+    * Must not be unit tests.
+    * Must not incorporate more than one subsystem or feature.
+
+5. Samples must be documented.
+++++++++++++++++++++++++++++++
+  * Samples must have a README.rst file in the samples folder.
+    Example: samples/subsys/foo/README.rst. clearly explaining the purpose of the sample, its
```

Comment:
Please use backticks for file names etc. all over. Example:

```suggestion
  * Samples must have a `README.rst` file in the samples folder.
    Example: `samples/subsys/foo/README.rst` clearly explaining the purpose of the sample, its
```

Reply:
> @aasthagr I think the overall changes in this PR make for a nice addition to the documentation. Do you think you can address the issues raised in time to get them into v3.2?

@henrikbrixandersen Missed the 3.2 RC1, maybe it can still make it to 3.2 as RC2 is 09/16. I have updated this PR as per the requested changes. You may re-review again now. Thanks!!